src/disk_interface.cc

```diff
@@ -221,7 +221,72 @@ TimeStamp RealDiskInterface::Stat(const string& path, string* err) const {
 #endif
 }
 
+#ifdef _WIN32
+bool UTF8ToWide(wstring& wide, const string& utf8) {
+  if (utf8.empty()) {
+    wide = L"";
+    return true;
+  } else {
+    vector<wchar_t> wdata(MultiByteToWideChar(CP_UTF8, 0, utf8.c_str(), utf8.size(), NULL, 0));
+    int dw = MultiByteToWideChar(CP_UTF8, 0, utf8.c_str(), utf8.size(), wdata.data(), wdata.size());
+    if (dw == 0) {
+      return false;
+    }
+    wide = wstring(wdata.data(), dw);
+    return true;
+  }
+}
+
+static bool fullPathName(wstring& result, const wstring& wpath) {
+  vector<wchar_t> buf(100);
+  DWORD dw = GetFullPathNameW(wpath.c_str(), buf.size(), buf.data(), NULL);
```

Comment:
As far as I know GetFullPathName is slow API, can I ask you to show this change does affect performance of ninja execution?

Reply:
I hope, as for an addition to slow ```CreateFileW```+```WriteFile``` the overhead should be neglible.
But if we prefix all the filesystem API (including ```StatFile``` and others), then there might be some space for optimization: caching results, checking for the cases when ```GetFullPathName``` can be safely skipped, etc.

Reply:
```GetFullPathName``` is to be replaced with a hand-written function anyway: it puts names like ```"nul"``` to "\\.\"-namespace instead of "\\?\"